project-koku/koku
# COST-2626: Fine tuning our GCP resummary range

Just testing in eph for now. Will update if solution works.

Fixes [COST-2626](https://issues.redhat.com/browse/COST-2626)

Changes proposed in this PR:
* Change the logic for how we decide if we should do a full monthly update or not.  

Testing Instructions:
1. You will need to test this with the day to day workflow. I have some notes written up about [how to do this here](https://github.com/myersCody/cody_notes/blob/main/gcp/day_to_day_flow.md#how-to-manually-test).

2. You will want your initial day to be:

3. After you have completed your initial download, you will change the data to:

4. After that you can hit the download url again and monitor the logs to see something like this:
```
koku-worker_1     | [2022-05-13 14:05:21,643] INFO 40fd3f5c-9644-4611-a4c8-3484a92d4d87 Updating GCP report summary tables from parquet:
koku-worker_1     | 	Schema: acct10001
koku-worker_1     | 	Provider: c92d75cf-697a-4402-a5a4-6e1d6c559251
koku-worker_1     | 	Dates: 2022-04-29 - 2022-05-01
koku-worker_1     | [2022-05-13 14:05:21,644] INFO 40fd3f5c-9644-4611-a4c8-3484a92d4d87 Deleting records from reporting_gcpcostentrylineitem_daily_summary for source c92d75cf-697a-4402-a5a4-6e1d6c559251 from 2022-04-29 to 2022-05-01
koku-worker_1     | [2022-05-13 14:05:21,644] INFO 40fd3f5c-9644-4611-a4c8-3484a92d4d87 Triggering DELETE on <class 'reporting.provider.gcp.models.GCPCostEntryLineItemDailySummary'> from 2022-04-29 to 2022-05-01.
koku-worker_1     | [2022-05-13 14:05:21,655] INFO 40fd3f5c-9644-4611-a4c8-3484a92d4d87 Finished DELETE on <class 'reporting.provider.gcp.models.GCPCostEntryLineItemDailySummary'> in 0.009792 seconds.
koku-worker_1     | [2022-05-13 14:05:24,633] INFO 40fd3f5c-9644-4611-a4c8-3484a92d4d87 reporting_gcpcostentrylineitem_daily_summary.sql for acct10001
```
^ Notice that the date range here is from `2022-04-29` to `2022-05-01`. 

---
Additional Context:
